src/Boost.sol

```diff
@@ -69,62 +74,71 @@ contract Boost is IBoost, EIP712("boost", "1"), Ownable {
+            // tokenFeeAmount < _amount, therefore balance will never underflow
             balance = _amount - tokenFeeAmount;
             tokenFeeBalances[address(_token)] += tokenFeeAmount;
         } else {
+            // When there is no token fee, the boost balance is full deposit amount
             balance = _amount;
         }
 
-        uint256 newId = nextBoostId;
-        nextBoostId++;
-        boosts[newId] = BoostConfig({
-            strategyURI: _strategyURI,
-            token: _token,
-            balance: balance,
-            guard: _guard,
-            start: _start,
-            end: _end,
-            owner: _owner
-        });
+        uint256 boostId = nextBoostId;
+        unchecked {
+            // Overflows if 2**256 boosts are minted
+            nextBoostId++;
+        }
 
+        // Minting the boost as an ERC721 and storing the config data
+        _safeMint(_owner, boostId);
+        _setTokenURI(boostId, _strategyURI);
+        boosts[boostId] = BoostConfig({ token: _token, balance: balance, guard: _guard, start: _start, end: _end });
+
+        // Transferring the deposit amount of the ERC20 token to the contract
         _token.transferFrom(msg.sender, address(this), _amount);
 
-        emit BoostCreated(newId, boosts[newId]);
+        emit BoostCreated(boostId, boosts[boostId]);
     }
 
     /// @notice Top up an existing boost
     function depositTokens(uint256 _boostId, uint256 _amount) external override {
+        BoostConfig storage boost = boosts[_boostId];
         if (_amount == 0) revert BoostDepositRequired();
-        if (boosts[_boostId].owner == address(0)) revert BoostDoesNotExist();
-        if (boosts[_boostId].end <= block.timestamp) revert BoostEnded();
+        if (!_exists(_boostId)) revert BoostDoesNotExist();
+        if (boost.end <= block.timestamp) revert BoostEnded();
 
         uint256 balanceIncrease = 0;
         if (tokenFee > 0) {
+            // The token fee is calculated and subtracted from the deposit amount to get the boost balance increase
             uint256 tokenFeeAmount = _amount / tokenFee;
             balanceIncrease = _amount - tokenFeeAmount;
-            tokenFeeBalances[address(boosts[_boostId].token)] += tokenFeeAmount;
+            tokenFeeBalances[address(boost.token)] += tokenFeeAmount;
         } else {
+            // When there is no token fee, the boost balance increase is the full deposit amount
             balanceIncrease = _amount;
         }
 
-        boosts[_boostId].balance += balanceIncrease;
-        boosts[_boostId].token.transferFrom(msg.sender, address(this), _amount);
+        boost.balance += balanceIncrease;
+        boost.token.transferFrom(msg.sender, address(this), _amount);
 
         emit TokensDeposited(_boostId, msg.sender, balanceIncrease);
     }
 
     /// @notice Withdraw remaining tokens from an expired boost
     function withdrawRemainingTokens(uint256 _boostId, address _to) external override {
```

Comment:
How about using `burn` function name here and burn the NFT? This is similar than what Uniswap is doing I assume with LP NFT position

Reply:
Maybe if boost is empty we can still use the same burn function but would not withdraw any token just burn the NFT

Reply:
And if no-one burn it, it still fine

Reply:
Okay ill add that functionality, it shouldn't add too much overhead